README.md

```diff
@@ -22,25 +24,126 @@ pip install framegrab
 
-To grab a frame, simply call the `grab()` method:
+Using a single webcam with several options would look something like this:
+```
+config = {
+    'name': 'front door camera',
+    'input_type': 'webcam',
+    'id': {
+        'serial_number': 23432570
+    },
+    'options': {
+        'resolution': {
+            'height': 1080,
+            'width': 1920,
+        },
+        'zoom': {
+            'digital': 1.5
+        }
+
+    }
+}
+
+grabber = FrameGrabber.create_grabber(config)
+```
 
+To get a frame, simply run:
 ```
 frame = grabber.grab()
-if frame is None:
-    print("No frame captured!")
+```
+You can also change the options after the grabber is created.
+```
+new_options = {
+    'resolution': {
+        'height': 480,
+        'width': 640,
+    },
+    'crop': {
+        'relative': {
+            'top': .1,
+            'bottom': .9,
+            'left': .1,
+            'right': .9,
+        }
+    }
+}
+
+grabber.apply_options(new_options)
+```
+
+When you are done with the camera, release the resource by running:
+```
+grabber.release()
+```
+
+You might have several cameras of different types that you want to use in the same application. In this case, you can load the configurations from a yaml file and use `FrameGrabber.create_grabbers`.
+
+If you have multiple cameras of the same type plugged in, it's recommended to provide serial numbers in your configurations; this will ensure that each configuration is paired with the correct camera. If you don't provide serial numbers in your configurations, configurations will be paired with cameras in a sequential manner. 
+
+Below is a sample yaml file containing configurations for three different cameras.
+```
+GL_CAMERAS: |
+  - name: on robot arm
+    input_type: realsense
+    options: 
+      depth:
+        side_by_side: 1
+      crop:
+        relative:
+          right: .8
+  - name: conference room
+      input_type: rtsp
+      address: 
+        rtsp_url: <YOUR RTSP URL>
+      options:
+        crop:
+          absolute:
+            top: 350
+            bottom: 1100
+            left: 1100
+            right: 2000
+  - name: workshop
+    input_type: webcam
+    id:
+      serial_number: B77D3A8F
+```
+You can load the configurations and use the cameras in the following manner.
+```
+from framegrab import FrameGrabber
+import yaml
+
+# load the configurations from yaml
+config_path = 'camera_config.yaml'
+with open(config_path, 'r') as f:
+    data = yaml.safe_load(f)
+    configs = yaml.safe_load(data['GL_CAMERAS'])
+
+# create the grabbers
+grabbers = FrameGrabber.create_grabbers(configs)
+
+for grabber in grabbers.values():
+    print(grabber.config)
+    frame = grabber.grab()
+    display_image(frame)
+    grabber.release()
+```
+It is also possible to call `FrameGrabber.autodiscover`. This will find all available cameras and return them with default configurations. This method does not work with RTSP cameras.
+```
+grabbers = FrameGrabber.autodiscover()
```

Comment:
Would be nice to show how you can discover the serial numbers after running `autodiscover` say with a script that does autodiscover and then prints out the serial numbers.  Otherwise I'd have no clue how to find my serial numbers.

Reply:
Serial number discover is currently in a private method, not exposed to the user. Also, it doesn't add the serial_number into the config when autodiscover is called. That being said, this could easily be improved in the next release, perhaps with a get_serial_numbers method, or something like that.